aten/src/ATen/Context.h

```diff
@@ -94,8 +96,22 @@ class TORCH_API Context {
       AT_ERROR(c10::DeviceTypeName(device_type), " device type not enabled.");
     }
   }
-  static bool isPinnedPtr(const void* data) {
-    return detail::getCUDAHooks().isPinnedPtr(data);
+  bool isPinnedPtr(
+      const void* data,
+      std::optional<DeviceType> device_type = std::nullopt) {
```

Comment:
Can you use `c10::DeviceType` here and below to avoid that issue?

Reply:
Have updated.